packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiException.java

```diff
@@ -1,33 +1,48 @@
 package software.amazon.jsii;
 
-/**
+/*
  * An error raised by the jsii runtime.
  */
-public final class JsiiException extends RuntimeException {
+public abstract class JsiiException extends RuntimeException {
     public static final long serialVersionUID = 1L;
 
-    /**
-     * Creates an exception.
-     * @param message The error message
-     */
-    JsiiException(final String message) {
-        super(message);
-    }
+    public static enum Type {
```

Comment:
I don't think this needs to/should be public. Package-private should be enough.

```suggestion
   static enum Type {
```